cmd/check_senders.cpp

```diff
@@ -243,147 +206,158 @@ void process_txs_for_signing(ChainConfig& config, uint64_t block_num, BlockBody&
             rlp::encode(rlp, txn, true, {});
         }
 
-        ethash::hash256 hash{keccak256(rlp)};
-        Recoverer::package rp{block_num, hash, x.recovery_id, txn.r, txn.s};
+        auto hash{keccak256(rlp)};
+        Recoverer::package rp{block_num, hash, x.recovery_id};
+        intx::be::unsafe::store(rp.signature, txn.r);
+        intx::be::unsafe::store(rp.signature + 32, txn.s);
         packages.push_back(rp);
     }
 }
 
 bool start_workers(std::vector<std::unique_ptr<Recoverer>>& workers) {
-    for (size_t r = 0; r < workers.size(); r++) {
-        std::cout << format_time() << " Starting worker thread #" << r << std::endl;
-        workers.at(r)->start();
+    for (const auto& worker : workers) {
+        SILKWORM_LOG(LogLevels::LogInfo) << "Starting worker thread #" << worker->get_id() << std::endl;
+        worker->start();
         // Wait for thread to init properly
         std::this_thread::sleep_for(std::chrono::milliseconds(5));
-        if (workers.at(r)->get_state() != Worker::WorkerState::kStarted) {
+        if (worker->get_state() != Worker::WorkerState::kStarted) {
             return false;
         }
     }
     return true;
 }
 
 void stop_workers(std::vector<std::unique_ptr<Recoverer>>& workers, bool wait) {
-    for (size_t r = 0; r < workers.size(); r++) {
-        if (workers.at(r)->get_state() == Worker::WorkerState::kStarted) {
-            std::cout << format_time() << " Stopping worker thread #" << r << std::endl;
-            workers.at(r)->stop(wait);
+    for (const auto& worker : workers) {
+        if (worker->get_state() == Worker::WorkerState::kStarted) {
+            SILKWORM_LOG(LogLevels::LogInfo) << "Stopping worker thread #" << worker->get_id() << std::endl;
+            worker->stop(wait);
         }
     }
 }
 
-std::optional<uint64_t> get_highest_canonical_header(std::unique_ptr<lmdb::Table>& headers) {
-    size_t count{0};
-    lmdb::err_handler(headers->get_rcount(&count));
-    if (!count) {
-        return std::nullopt;
-    }
+uint64_t load_canonical_headers(std::unique_ptr<lmdb::Transaction>& txn, uint64_t from, uint64_t to, evmc::bytes32* out) {
```

Comment:
Since this method simply uses txn and doesn't assume ownership, it's faster and simpler just to pass
`lmdb::Transaction*` or `lmdb::Transaction&` instead of `std::unique_ptr<lmdb::Transaction>&`.

Also, I'd change the output parameter to `std::vector<evmc::bytes32>& out` in order to avoid C-style memory management.